src/include/pmix_globals.c

```diff
@@ -402,7 +402,7 @@ void pmix_execute_epilog(pmix_epilog_t *epi)
          * matches that of the peer - we do this to provide at least
          * some minimum level of protection */
         tmp = pmix_argv_split(cf->path, ',');
-        for (n=0; NULL != tmp[n]; n++) {
+        for (n=0; NULL != tmp && NULL != tmp[n]; n++) {
```

Comment:
It isn't possible for tmp to be NULL - if there were no epilog, then there wouldn't be an entry on the list. What am I missing?

Reply:
I used the latest `master` branches from open-mpi and pmix, use pmix as an external component (both configure'd with `--disable-dlopen`, not sure this is relevant though) and `mpirun` crashed.
This patch fixed the issue and I did not dig on how we ended up here.

Reply:
We should find the root cause instead, then, as this should not be possible. It sounds like either someone is registering a NULL epilog path (which is an error), or something is getting destructed too soon.

Reply:
could there be a typo at line 404 and we should `pmix_argv_split(cd->path, ...) instead (e.g. `cd` instead of `cf` which is a loop invariant) ?

Reply:
Perhaps we should add a check in the job control code for registering epilogs - if someone gives us a NULL path, then we should return an error. Might help us spot the source.

Reply:
Ah - indeed!! A copy/paste error! Good eyes - that should definitely be the fix

Reply:
I will try this and update the PR